Source/Core/VideoCommon/Assets/DirectFilesystemAssetLibrary.cpp

```diff
@@ -488,8 +488,6 @@ bool DirectFilesystemAssetLibrary::LoadMips(const std::filesystem::path& asset_p
 
     data->m_levels.push_back(std::move(level));
   }
-
-  return true;
```

Comment:
I don't understand this change. (Which also means that even if it is somehow correct to remove it, it should probably stay anyway to avoid confusing people.)

Reply:
The loop before it is an infinite loop, and because it uses an unsigned int it also doesn't invoke UB (and is guaranteed to terminate at some point.) So any compiler is likely to accept this as not being present.

(to be fair, I wanted to remark the same thing and refrained from doing it, after finding _that_ out. But I wonder if it would make more sense to terminate the loop in a more obvious way instead...)

Reply:
What is UB in this context and how do we terminate this in a more obvious way?

Reply:
Undefined Behavior, something the C++ standard doesn't describe in detail and allows a compiler to do whatever it wants. Unsigned overflow is defined by the standard (so this loop is fine,) but signed overflow wouldn't be (leading to a potential infinite loop, ignored loop, empty method, or formatting your hard drive. Nobody knows.)

Terminate it more obviously? Probably by picking an arbitrary upper limit for the mips, and putting that into the loop condition. Not sure what that number would be, though.

Reply:
`std::numeric_limits<unsigned int>::max()` or `std::numeric_limits<unsigned int>::infinity()`, maybe?

Reply:
Yeah no, way less than that in my head. I can't feasibly see a mip level above....i dunno, 100? 1000? 10000?

Reply:
A mipmap level is always half the size (rounded up) of the previous level until both width and height reach 1. So in practice you're limited by the maximum texture size of the graphics API/driver/GPU, which is probably like 32k or so on current GPUs, and I wouldn't expect them to get too much higher in the future but who knows about that. So yeah, 100 is more than enough for anything realistic.

Reply:
I'm not sure how I should write that. My current iteration is `for (u32 initial_mip_level = static_cast<u32>(data->m_levels.size()), mip_level = initial_mip_level; mip_level < initial_mip_level + 100; mip_level++)`, which is quite ugly but leaves the surrounding code as untouched as possible.

Reply:
Am I alone in thinking this removal is fine as-is? The compiler will warn when a non-void function does not return a value in all control paths, and in this case no warning is emitted.

Reply:
I had to do a double-take to see it is in-fact an infinite loop, so I'm a bit torn about it. No preference either way though, leaning towards "get rid of the infinite loop" if I had to pick.

Reply:
Why is it so important that this for loop terminate eventually by condition? Is there an unspoken concern that allowing any number of mipmaps is a potential attack vector? An "infinite" loop which processes data until there is nothing left is not an uncommon idiom.

Reply:
Not saying it has to, but if we insist on changing anything in here, that would be my suggestion. Mainly because it seems incredibly unliklely to have a ridiculously high mip level (even when the only effect would be a local delay until loading completes if someone really went through the trouble of creating a ton of files.)

Due to the loop size (which can fit the screen, but needs expanding if viewed in a diff like this) it isn't immediately obvious that it is in fact an infinite loop that normally terminates very quickly for the absence of a file; and while it is correct that the `return` statement is technically unreachable, it likely raises more questions when reading the code (as is evident by this long thread already.)

But, again, I don't see the need to do anything here and just leave the code as-is.

Reply:
I'd probably rather leave this code as is, though I see why it was flagged.  I've never liked this code but I think Android currently needs PNG mips since it doesn't support DDS in some cases (maybe we could switch to something else).

If we want to avoid this "infinite loop", a more appropriate way would be to loop over the files in the filesystem imo.

But if we really want to use this as is, I'd like a comment above the 100 part, as it seems odd to use the 100.